pcp/versions/2.0/versioning.md

```diff
@@ -0,0 +1,94 @@
+Message Versioning
+===
+
+This section describes how the Puppet Communications Protocol handles clients
+attempting to communicate using different versions of the protocol. Here, we
+will assume that WebSocket provides the wire communication layer; a WebSocket
+URI will convey the version information when a PCP connection is attempted.
+
+### How clients and the pcp-broker determine protocol version
+
+When a pcp-broker starts up it will mount a URI for each of the
+versions it supports. Consider the following scenario:
+
+A pcp-broker, broker.example.com, supports versions 1, 2 and 3 of the PCP. It
+will mount the following URI's.
+
+- wss://broker.example.com/v1
+- wss://broker.example.com/v2
+- wss://broker.example.com/v3
```

Comment:
Hmm, I wonder if we want to keep this, in which case the URI including the client type would look like:
```
wss://broker.example.com/pcp/v2/controller
```
or
```
wss://broker.example.com/pcp/v2?type=controller
```
or we rather want to go (if at all possible) with the more flexible:
```
wss://broker.example.com/pcp?version=2.0&type=controller
```